state/runtime/evm/instructions.go

```diff
@@ -1191,10 +1191,16 @@ func (c *state) buildCallContract(op OpCode) (*runtime.Contract, uint64, uint64,
 	addr, _ := c.popAddr()
 
 	var value *big.Int
-	if op == CALL || op == CALLCODE {
+	if op == CALL || op == CALLCODE || op == AUTHCALL {
 		value = c.pop()
 	}
 
+	var extValue *big.Int
+	if op == AUTHCALL {
+		extValue = c.pop() // todo: propagate the extValue
+		_ = extValue       // ignore for now
```

Comment:
The EIP specifies a behaviour for this value:
```
If valueExt is not zero, the instruction immediately returns 0. In this case the gas that would have been passed into the call is refunded, but not the gas consumed by the AUTHCALL opcode itself. In the future, this restriction may be relaxed to externally transfer value out of the authorized account.
```

Reply:
Is this correct fix https://github.com/0xPolygon/polygon-edge/pull/1291/commits/44f81073acd1f6fd4d2dfba449b4ba2f962a8771 ?